### ISSUE TITLE
[BLOCKER] : All leaf devices telemetry (d2c message) stops working if we deactivate any leaf device (Transparent gateway scenario)

### ISSUE THREAD

**Issue by niravjprabtani** [Azure/iotedge#4354]
## Expected Behavior
Only the deactivated leaf device's telemetry should stop and the remaining leaf device should work as it is.

## Current Behavior
All leaf devices associated with the gateway stops sending telemetry if any of the device is deactivated.

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. Associate multiple leaf device with the gateway (Parent/Child association, Transparent gateway scenario)
2. Mark any one leaf device inactive from the azure portal 
3. Check for telemetry (device to cloud messages) using device explorer for remaining devices
4. Just activate that device and everything will work as expected

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```

√ config.yaml is well-formed - OK
√ config.yaml has well-formed connection string - OK
√ container engine is installed and functional - OK
√ Windows host version is supported - OK
√ config.yaml has correct hostname - OK
√ config.yaml has correct URIs for daemon mgmt endpoint - OK
√ latest security daemon - OK
√ host time is close to real time - OK
√ container time is close to host time - OK
‼ DNS server - Warning
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
√ production readiness: certificates - OK
√ production readiness: container engine - OK
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its C:\Windows\Temp\edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
√ production readiness: Edge Hub's storage directory is persisted on the host filesystem - OK

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with DPS endpoint - OK
√ host can connect to and perform TLS handshake with IoT Hub AMQP port - OK
√ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with IoT Hub MQTT port - OK
√ container on the IoT Edge module network can connect to IoT Hub AMQP port - OK
√ container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the IoT Edge module network can connect to IoT Hub MQTT port - OK

19 check(s) succeeded.
3 check(s) raised warnings. Re-run with --verbose for more details.
PS C:\nirav>




```
</details>

### Device Information
* Host OS [Windows 10 version 1809]: 
* Container OS [Windows containers]: 

### Runtime Versions
* aziot-edged [iotedge 1.0.10.4 (57772714c81c8b823a5ef05bf11bf343b923fb6a)]: 

## Logs

<details>
<summary>edgeHub logs in normal condition</summary>

```

<6> 2021-02-05 06:49:59.212 -05:00 [INF] - Updated reported properties for 19e0f9a4f3b542e881051e95c664b35a/ddiotdevicehostmodule
<6> 2021-02-05 06:49:59.571 -05:00 [INF] - Updated reported properties for cf72a8304c7d4b66b218b96f72effcb5
<6> 2021-02-05 06:50:03.115 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/ddiotdevicehostmodule
<6> 2021-02-05 06:50:06.266 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/ddiotdevicehostmodule
<6> 2021-02-05 06:50:17.396 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/ddiotedgeremoteaccessmodule
<6> 2021-02-05 06:50:19.390 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/mipredictiveinsightsmodule
<6> 2021-02-05 06:50:19.551 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:50:20.003 -05:00 [INF] - Updated reported properties for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:50:21.310 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/ddiotedgeremoteaccessmodule
<6> 2021-02-05 06:50:22.423 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:50:22.737 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/mipredictiveinsightsmodule
<6> 2021-02-05 06:50:25.281 -05:00 [INF] - Processing pending subscriptions for cf72a8304c7d4b66b218b96f72effcb5
<6> 2021-02-05 06:50:25.416 -05:00 [INF] - Updated reported properties for 19e0f9a4f3b542e881051e95c664b35a/ddiotedgeremoteaccessmodule
<6> 2021-02-05 06:50:25.787 -05:00 [INF] - Updated reported properties for 19e0f9a4f3b542e881051e95c664b35a/mipredictiveinsightsmodule
<6> 2021-02-05 06:50:26.563 -05:00 [INF] - Processing pending subscriptions for 19e0f9a4f3b542e881051e95c664b35a/ddiotdevicehostmodule
<6> 2021-02-05 06:50:28.318 -05:00 [INF] - Processing pending subscriptions for cf72a8304c7d4b66b218b96f72effcb5
<6> 2021-02-05 06:50:31.180 -05:00 [INF] - Updated reported properties for 19e0f9a4f3b542e881051e95c664b35a/ddiotdevicehostmodule
<6> 2021-02-05 06:50:31.586 -05:00 [INF] - Updated reported properties for cf72a8304c7d4b66b218b96f72effcb5
<6> 2021-02-05 06:50:49.701 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6

```
</details>

<details>
<summary>edgeHub logs when one device is deactivated</summary>

```

<6> 2021-02-05 06:52:30.865 -05:00 [INF] - Connection status for 99d1f2b47baa4521a081f5af530d58e6 changed to DisconnectedTokenExpired
<6> 2021-02-05 06:52:30.865 -05:00 [INF] - Removing device connection for device 99d1f2b47baa4521a081f5af530d58e6 with removeCloudConnection flag 'True'.
<6> 2021-02-05 06:52:30.865 -05:00 [INF] - Closing AMQP device proxy for 99d1f2b47baa4521a081f5af530d58e6 because no handler was registered.
Microsoft.Azure.Devices.Edge.Hub.Core.EdgeHubConnectionException: Connection closed for device 99d1f2b47baa4521a081f5af530d58e6.
<6> 2021-02-05 06:52:30.875 -05:00 [INF] - Closing link Events for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link Events for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link TwinSending for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link TwinSending for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link TwinReceiving for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link TwinReceiving for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link MethodSending for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.877 -05:00 [INF] - Closing link MethodSending for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.878 -05:00 [INF] - Closing link MethodReceiving for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.878 -05:00 [INF] - Closing link MethodReceiving for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.878 -05:00 [INF] - Closing link C2D for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.878 -05:00 [INF] - Closing link C2D for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:30.878 -05:00 [INF] - Closing receiver in cloud proxy a0402cea-9121-4eb4-868e-fc058b60cd85 for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.047 -05:00 [INF] - New token received on the Cbs link
<6> 2021-02-05 06:52:31.047 -05:00 [INF] - Token updated for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.053 -05:00 [INF] - Opened link Events for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.064 -05:00 [INF] - Opened link MethodSending for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.065 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.066 -05:00 [INF] - Attempting to connect to IoT Hub for client 99d1f2b47baa4521a081f5af530d58e6 via AMQP...
<6> 2021-02-05 06:52:31.074 -05:00 [INF] - Opened link MethodReceiving for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.075 -05:00 [INF] - Opened link TwinSending for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.075 -05:00 [INF] - Opened link TwinReceiving for 99d1f2b47baa4521a081f5af530d58e6
<4> 2021-02-05 06:52:31.387 -05:00 [WRN] - Error creating cloud connection for client 99d1f2b47baa4521a081f5af530d58e6
Microsoft.Azure.Devices.Client.Exceptions.UnauthorizedException: error(condition:amqp:unauthorized-access,description:{"errorCode":401002,"trackingId":"9d4df566d62745388860b775a0f4ff3c-G:5-TimeStamp:02/05/2021 11:52:30","message":"Unauthorized","timestampUtc":"2021-02-05T11:52:30.4883057Z"},info:[com.microsoft:is-filtered:True])
 ---> Microsoft.Azure.Amqp.AmqpException: {"errorCode":401002,"trackingId":"9d4df566d62745388860b775a0f4ff3c-G:5-TimeStamp:02/05/2021 11:52:30","message":"Unauthorized","timestampUtc":"2021-02-05T11:52:30.4883057Z"}
   at Microsoft.Azure.Amqp.AsyncResult.End[TAsyncResult](IAsyncResult result)
   at Microsoft.Azure.Amqp.AmqpObject.OpenAsyncResult.End(IAsyncResult result)
   at Microsoft.Azure.Amqp.AmqpObject.EndOpen(IAsyncResult result)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenSendingAmqpLinkAsync(DeviceIdentity deviceIdentity, AmqpSession amqpSession, Nullable`1 senderSettleMode, Nullable`1 receiverSettleMode, String deviceTemplate, String moduleTemplate, String linkSuffix, String correlationId, TimeSpan timeout)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenSendingAmqpLinkAsync(DeviceIdentity deviceIdentity, AmqpSession amqpSession, Nullable`1 senderSettleMode, Nullable`1 receiverSettleMode, String deviceTemplate, String moduleTemplate, String linkSuffix, String correlationId, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenTelemetrySenderLinkAsync(DeviceIdentity deviceIdentity, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpUnit.EnsureSessionAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpUnit.OpenAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpTransportHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ProtocolRoutingDelegatingHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.<>c__DisplayClass23_0.<<ExecuteWithErrorHandlingAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation)
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass33_0.<<OpenInternalAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.EnsureOpenedAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.OpenAsync()
   at Microsoft.Azure.Devices.Edge.Util.TaskEx.TimeoutAfter(Task task, TimeSpan timeout) in C:\agent\_work\9\s\edge-util\src\Microsoft.Azure.Devices.Edge.Util\TaskEx.cs:line 144
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.DeviceClientWrapper.OpenAsync() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\DeviceClientWrapper.cs:line 59
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.<>c__DisplayClass30_0.<<InvokeFunc>b__0>d.MoveNext() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 187
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.InvokeFunc[T](Func`1 func, String operation, Boolean useForConnectivityCheck) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 180
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.InvokeFunc[T](Func`1 func, String operation, Boolean useForConnectivityCheck) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 180
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.OpenAsync() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 64
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnection.ConnectToIoTHub(ITokenProvider newTokenProvider) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnection.cs:line 134
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnection.CreateNewCloudProxyAsync(ITokenProvider newTokenProvider) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnection.cs:line 117
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnection.Create(IIdentity identity, Action`2 connectionStatusChangedHandler, ITransportSettings[] transportSettings, IMessageConverterProvider messageConverterProvider, IClientProvider clientProvider, ICloudListener cloudListener, ITokenProvider tokenProvider, TimeSpan idleTimeout, Boolean closeOnIdleTimeout, TimeSpan operationTimeout, String productInfo, Option`1 modelId) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnection.cs:line 99
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnectionProvider.<>c__DisplayClass17_1.<<Connect>b__2>d.MoveNext() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnectionProvider.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnectionProvider.Connect(IIdentity identity, Action`2 connectionStatusChangedHandler) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnectionProvider.cs:line 178
<6> 2021-02-05 06:52:31.414 -05:00 [INF] - Error getting cloud connection for device 99d1f2b47baa4521a081f5af530d58e6
Microsoft.Azure.Devices.Client.Exceptions.UnauthorizedException: error(condition:amqp:unauthorized-access,description:{"errorCode":401002,"trackingId":"9d4df566d62745388860b775a0f4ff3c-G:5-TimeStamp:02/05/2021 11:52:30","message":"Unauthorized","timestampUtc":"2021-02-05T11:52:30.4883057Z"},info:[com.microsoft:is-filtered:True])
 ---> Microsoft.Azure.Amqp.AmqpException: {"errorCode":401002,"trackingId":"9d4df566d62745388860b775a0f4ff3c-G:5-TimeStamp:02/05/2021 11:52:30","message":"Unauthorized","timestampUtc":"2021-02-05T11:52:30.4883057Z"}
   at Microsoft.Azure.Amqp.AsyncResult.End[TAsyncResult](IAsyncResult result)
   at Microsoft.Azure.Amqp.AmqpObject.OpenAsyncResult.End(IAsyncResult result)
   at Microsoft.Azure.Amqp.AmqpObject.EndOpen(IAsyncResult result)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenSendingAmqpLinkAsync(DeviceIdentity deviceIdentity, AmqpSession amqpSession, Nullable`1 senderSettleMode, Nullable`1 receiverSettleMode, String deviceTemplate, String moduleTemplate, String linkSuffix, String correlationId, TimeSpan timeout)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenSendingAmqpLinkAsync(DeviceIdentity deviceIdentity, AmqpSession amqpSession, Nullable`1 senderSettleMode, Nullable`1 receiverSettleMode, String deviceTemplate, String moduleTemplate, String linkSuffix, String correlationId, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenTelemetrySenderLinkAsync(DeviceIdentity deviceIdentity, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpUnit.EnsureSessionAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpUnit.OpenAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpTransportHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ProtocolRoutingDelegatingHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.<>c__DisplayClass23_0.<<ExecuteWithErrorHandlingAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation)
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass33_0.<<OpenInternalAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.EnsureOpenedAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.OpenAsync()
   at Microsoft.Azure.Devices.Edge.Util.TaskEx.TimeoutAfter(Task task, TimeSpan timeout) in C:\agent\_work\9\s\edge-util\src\Microsoft.Azure.Devices.Edge.Util\TaskEx.cs:line 144
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.DeviceClientWrapper.OpenAsync() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\DeviceClientWrapper.cs:line 59
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.<>c__DisplayClass30_0.<<InvokeFunc>b__0>d.MoveNext() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 187
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.InvokeFunc[T](Func`1 func, String operation, Boolean useForConnectivityCheck) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 180
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.InvokeFunc[T](Func`1 func, String operation, Boolean useForConnectivityCheck) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 180
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.OpenAsync() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 64
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnection.ConnectToIoTHub(ITokenProvider newTokenProvider) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnection.cs:line 134
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnection.CreateNewCloudProxyAsync(ITokenProvider newTokenProvider) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnection.cs:line 117
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnection.Create(IIdentity identity, Action`2 connectionStatusChangedHandler, ITransportSettings[] transportSettings, IMessageConverterProvider messageConverterProvider, IClientProvider clientProvider, ICloudListener cloudListener, ITokenProvider tokenProvider, TimeSpan idleTimeout, Boolean closeOnIdleTimeout, TimeSpan operationTimeout, String productInfo, Option`1 modelId) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnection.cs:line 99
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnectionProvider.<>c__DisplayClass17_1.<<Connect>b__2>d.MoveNext() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnectionProvider.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.CloudConnectionProvider.Connect(IIdentity identity, Action`2 connectionStatusChangedHandler) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\CloudConnectionProvider.cs:line 178
<6> 2021-02-05 06:52:31.422 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6, but no cloud proxy was found
<6> 2021-02-05 06:52:31.422 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.422 -05:00 [INF] - Processing pending subscriptions for 99d1f2b47baa4521a081f5af530d58e6
<6> 2021-02-05 06:52:31.877 -05:00 [INF] - Attempting to connect to IoT Hub for client 99d1f2b47baa4521a081f5af530d58e6 via AMQP...
<4> 2021-02-05 06:52:32.222 -05:00 [WRN] - Error creating cloud connection for client 99d1f2b47baa4521a081f5af530d58e6
Microsoft.Azure.Devices.Client.Exceptions.UnauthorizedException: error(condition:amqp:unauthorized-access,description:{"errorCode":401002,"trackingId":"2da5e3d51e234146ba742536db1d52e6-G:5-TimeStamp:02/05/2021 11:52:31","message":"Unauthorized","timestampUtc":"2021-02-05T11:52:31.3033492Z"},info:[com.microsoft:is-filtered:True])
 ---> Microsoft.Azure.Amqp.AmqpException: {"errorCode":401002,"trackingId":"2da5e3d51e234146ba742536db1d52e6-G:5-TimeStamp:02/05/2021 11:52:31","message":"Unauthorized","timestampUtc":"2021-02-05T11:52:31.3033492Z"}
   at Microsoft.Azure.Amqp.AsyncResult.End[TAsyncResult](IAsyncResult result)
   at Microsoft.Azure.Amqp.AmqpObject.OpenAsyncResult.End(IAsyncResult result)
   at Microsoft.Azure.Amqp.AmqpObject.EndOpen(IAsyncResult result)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenSendingAmqpLinkAsync(DeviceIdentity deviceIdentity, AmqpSession amqpSession, Nullable`1 senderSettleMode, Nullable`1 receiverSettleMode, String deviceTemplate, String moduleTemplate, String linkSuffix, String correlationId, TimeSpan timeout)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenSendingAmqpLinkAsync(DeviceIdentity deviceIdentity, AmqpSession amqpSession, Nullable`1 senderSettleMode, Nullable`1 receiverSettleMode, String deviceTemplate, String moduleTemplate, String linkSuffix, String correlationId, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpIoTSession.OpenTelemetrySenderLinkAsync(DeviceIdentity deviceIdentity, TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpUnit.EnsureSessionAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.AmqpIoT.AmqpUnit.OpenAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Transport.Amqp.AmqpTransportHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ProtocolRoutingDelegatingHandler.OpenAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.<>c__DisplayClass23_0.<<ExecuteWithErrorHandlingAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation)
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass33_0.<<OpenInternalAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.EnsureOpenedAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.OpenAsync()
   at Microsoft.Azure.Devices.Edge.Util.TaskEx.TimeoutAfter(Task task, TimeSpan timeout) in C:\agent\_work\9\s\edge-util\src\Microsoft.Azure.Devices.Edge.Util\TaskEx.cs:line 144
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.DeviceClientWrapper.OpenAsync() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\DeviceClientWrapper.cs:line 59
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.<>c__DisplayClass30_0.<<InvokeFunc>b__0>d.MoveNext() in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 187
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Edge.Hub.CloudProxy.ConnectivityAwareClient.InvokeFunc[T](Func`1 func, String operation, Boolean useForConnectivityCheck) in C:\agent\_work\9\s\edge-hub\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ConnectivityAwareClient.cs:line 180

```
</details>

## Additional Information
Please provide any additional information that may be helpful in understanding the issue.

There are two leaf devices and deviceId are as mentioned below

1) 99d1f2b47baa4521a081f5af530d58e6
2) cf72a8304c7d4b66b218b96f72effcb5

i.e. : If I disable first device (99d1f2b47baa4521a081f5af530d58e6) then telemetry stops working of device cf72a8304c7d4b66b218b96f72effcb5 too.

Note: I am checking telemetry using device explorer exe.

**Comment by niravjprabtani** [comment]
In addition to that edgeHub stops processing device to cloud message for all associated devices if there is any connection issue in one of the devices.

If it will stop for a particular device that is having an issue then I can understand, but why it stops processing telemetry for all devices?

**Comment by ancaantochi** [comment]
All messages to the cloud are using the same queue and when a leaf device is disabled will cause issues for all telemetry. We are working on fixing the issue.

**Comment by niravjprabtani** [comment]
Thanks for the update, We are waiting for the resolution of the same.

**Comment by ancaantochi** [comment]
Fixed by 756b83b9062624a3d0c2e55f59a56195cc6704b8 and released in 1.1.1
